### PR TITLE
feat: Update of the Book→Operators page

### LINKED ISSUE #178: Incorrect description of the division operator (`/`)
> used for integer division of two values, which truncates towards zero

`/` does not truncate towards zero, it actually behaves like the `floor` function, i.e. it truncates towards zero for the positive results and _away_ from zero for the negative results.

### LINKED ISSUE #181: Mention `i / j * j + i % j == i` holds in book/operators

### LINKED ISSUE #192: Operator precedence levels are not documented
We model it after JS/TS: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence

### LINKED ISSUE #202: `book/operators` `logical OR, ||` section has incorrect example
An example from `logical OR, || ` section is a copy paste from `logical AND, && `, should be corrected

### PR DESCRIPTION
**Pull request by novusnota**
* All precedences match Tact 1.3.0
* Added bitwise XOR operator `^`
* Fixed minor typos and corrected some descriptions
* Added a table of operators in decreasing order of their precedences

Note, that the only link leading to Language section now has a proper link to the Reference section. This is intentional, as the Lang->Ref rename is almost there :)

Closes #192 (new operator precedence + a table) — this is the main issue target of this PR
Closes #202 (small error)
Closes #181 (small note)
Closes #178 (small correction)

### CODE REVIEW

File: pages/book/operators.mdx [hunk 1]
```diff
@@ -27,7 +84,15 @@ Consider the following code:
 
 Even though this example may be simple, neglection of precedence rules can often lead to confusing situations with operators. The correct order of operations can be ensured by wrapping every operation in [parentheses](#parentheses), since parentheses have the highest precedence of all expressions and operators there is.
 
-## Parentheses, `()`
+That said, Tact's precedence levels are carefully crafted with predictability and convenience in mind, such that the following expression will always hold `true{:tact}`:
+
+```tact
+a / b * b + a % b == a; // true for any Int values of `a` and `b`,
```
Comment (anton-trunov) [inline pages/book/operators.mdx:90]: This is not about operator precedence level, it's a basic property of division. This should be moved to the corresponding section

File: pages/book/operators.mdx [hunk 2]
```diff
@@ -98,24 +163,27 @@ Can only be applied to values of type [`Int{:tact}`][int]:
 
 ```tact
 let two: Int = 2;
-two * two;         // 4
-0 * 1_000_000_000; // 0
--1 * 5;            // -5
+two * two;           // 4
+ 0  * 1_000_000_000; // 0
+-1  * 5;             // -5
```
Comment (anton-trunov) [inline pages/book/operators.mdx:168]: why this change?

File: pages/book/operators.mdx [hunk 3]
```diff
@@ -98,24 +163,27 @@ Can only be applied to values of type [`Int{:tact}`][int]:
...
 
 pow(2, 255) * pow(2, 255); // build error: integer overflow!
 ```
 
 #### Divide, `/` [#binary-divide]
 
-Binary slash (_division_) operator `/{:tact}` is used for integer division of two values, which truncates towards zero. An attempt to divide by zero would result in an error with [exit code 4](/book/exit-codes#4): `Integer overflow`.
+Binary slash (_division_) operator `/{:tact}` is used for integer division of two values, which truncates towards zero if result is positive, and away from zero if result is negative. An attempt to divide by zero would result in an error with [exit code 4](/book/exit-codes#4): `Integer overflow`.
```
Comment (anton-trunov) [inline pages/book/operators.mdx:175]: We can also mention this is called "rounding to −∞"

File: pages/book/operators.mdx [hunk 4]
```diff
@@ -347,46 +409,80 @@ nullable == anotherNullable; // false
...
 
-Binary _less than or equal to_ operator `<={:tact}` returns `true{:tact}` if the left operand is less than or equal to the right operand, and `false{:tact}` otherwise. Can only be applied to values of type [`Int{:tact}`][int]:
+  [Bitwise XOR - Wikipedia](https://en.wikipedia.org/wiki/Bitwise_operation#XOR)\
+  [Bit manipulation - Wikipedia](https://en.wikipedia.org/wiki/Bit_manipulation)
+
+</Callout>
+
+### Bitwise OR, `|` [#binary-bitwise-or]
+
+Binary bar (_bitwise OR_) operator `|{:tact}` applies a [bitwise OR](https://en.wikipedia.org/wiki/Bitwise_operation#OR), which performs the [logical OR](#binary-logical-or) operation on each pair of the corresponding bits of operands. This is useful when we want to apply a specific [bitmask](https://en.wikipedia.org/wiki/Mask_(computing)).
```
Comment (anton-trunov) [inline pages/book/operators.mdx:464]: Speaking of bit masks, here would be a nice place to link to the sendmodes combinations section

**Review by anton-trunov — CHANGES_REQUESTED** [review]

File: pages/book/operators.mdx (hunk 2 above)
Comment (novusnota) [reply pages/book/operators.mdx:168]: Felt like aligning things around the operator would make for better readability at a glance. May remove this change here and in other places across the file as well

Comment (novusnota) [reply pages/book/operators.mdx:168]: Makes sense, will revert those changes

File: pages/book/operators.mdx [hunk 5]
```diff
@@ -107,17 +164,36 @@ pow(2, 255) * pow(2, 255); // build error: integer overflow!
 
 #### Divide, `/` [#binary-divide]
 
-Binary slash (_division_) operator `/{:tact}` is used for integer division of two values, which truncates towards zero. An attempt to divide by zero would result in an error with [exit code 4](/book/exit-codes#4): `Integer overflow`.
+Binary slash (_division_) operator `/{:tact}` is used for integer division of two values, which truncates towards zero if result is positive, and away from zero if result is negative. This is also called [rounding down](https://en.wikipedia.org/wiki/Rounding#Rounding_down) (or rounding towards $-∞$), where, for example, result value of $23.7$ gets rounded to $23$, and $−23.2$ gets rounded to $−24$.
```
Comment (anton-trunov) [inline pages/book/operators.mdx:167]: It's confusing to mention real numbers in the context of integer division

File: pages/book/operators.mdx [hunk 6]
```diff
@@ -107,17 +164,36 @@ pow(2, 255) * pow(2, 255); // build error: integer overflow!
...
+-1 / -5; // 0
+1 / -5;  // -1
+1 / 5;   // 0
+6 / 5;   // 1, rounding down
+-6 / 5;  // -2, rounding down (towards -∞)
 ```
 
+<Callout>
+
+  Because division operator `/{:tact}` in Tact rounds down, the following expression always holds `true{:tact}`:
```
Comment (anton-trunov) [inline pages/book/operators.mdx:187]:
```suggestion
  Note the following relationship between the division and modulo operators always holds:
```

**Review by anton-trunov — CHANGES_REQUESTED** [review]

File: pages/book/operators.mdx (hunk 6 above)
Comment (novusnota) [reply pages/book/operators.mdx:187]: If division operator wouldn't round down and produced real numbers, that wouldn't work. Euclidean-ish thingy you're suggesting only works with Ints. I may rewrite that part a bit different, mentioning that Tact & TON only operate on integer numbers

**Review by anton-trunov — APPROVED** [review]
